### PR TITLE
Added Northern Ireland Demographics (Local Authorities & Health View)

### PR DESCRIPTION
**Pull request by MatteoLarrode**
Hey Mike, I added demographics data for Northern Ireland, using contributions from Niuni. 

**Local Authorities**: from demographr, I used Niuni's Age and Gender data from at ltla level, and my ethnicity data at ltla level too. Both from the NISRA 2021 Census 

I want to mention something regarding the aggregation of ethnicities into higher level categories. NISRA does not specify any high level categories like the ONS does. So I used my own judgement, with help from Adja, to get as close as possible to ONS categories, with the objective to be consistent with the rest of the app.

**Health and Social Care Trusts**: Niuni added HSCT level age and gender data to demographr, which I used. 
However, I could not find ethnicity data at Health level, so I aggregated ltla ethnicity metrics to health level. I was careful not to aggregate the percentages: I used total populations (specified in the same dataset as ethnicity data) to recalculate the percentages at health level. 

I chose NOT to create it in a new demographr dataset because ethnicity data at health level was calculated by me and not pulled from an official source. 

---

_Last question_: I used the {httr} package for req_perform (when downloading an excel file from a URL). Does this create a dependency? If so, should it be added to the description? 
Or is it just used once to create the data and is not included as a dependency?

I would guess the latter because data-raw/ scripts are not ran when the app is running, but only when creating corresponding datasets.

### CODE REVIEW

**Comment by MikeJohnPage** [comment]
> So I used my own judgement, with help from Adja, to get as close as possible to ONS categories, with the objective to be consistent with the rest of the app.

Good call.

>  was careful not to aggregate the percentages: I used total populations (specified in the same dataset as ethnicity data) to recalculate the percentages at health level.

Again, good call!

> I chose NOT to create it in a new demographr dataset because ethnicity data at health level was calculated by me and not pulled from an official source.

A third good call. We want to keep the humaniverse packages as close to source as possible.

> Last question: I used the {httr} package for req_perform (when downloading an excel file from a URL). Does this create a dependency? If so, should it be added to the description?

Yes, every new package used creates a dependency. The nuance is to understand where the dependency is created. Packages used in `data-raw/` do **not** create dependencies in the Shiny app, as this folder is not sourced when running the app with `pkgload::load_all()` in `app.R`. This means only packages used in files which are sourced by the app should be counted as dependencies to the app, and added to `DESCRIPTION`.

In any case, we should still be careful about adding new dependencies, as the files in `data-raw` will likely need to be run again in the future. All the same rules apply to selecting new packages (e.g., the package should be dependable, it should be tracked with `renv`, etc.).

File: data-raw/northern-ireland-hsct-demographics.R [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    number__working_age_females = working_age_females,
+    number__older_females = older_females,
+    number__younger_males = younger_males,
+    number__working_age_males = working_age_males,
+    number__older_males = older_males
+  ) |>
+  pivot_longer(!area_name,
+    names_to = c(".value", "variable"),
+    names_sep = "__"
+  ) |>
```
Comment (MikeJohnPage) [inline data-raw/northern-ireland-hsct-demographics.R:30]: This is great.

File: data-raw/northern-ireland-hsct-demographics.R [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    variable = case_when(
+      variable == "older_females" ~ "Older \nfemales (65+)",
+      variable == "working_age_females" ~ "Working age \nfemales (16-64)",
+      variable == "younger_females" ~ "Younger \nfemales (< 16)",
+      variable == "older_males" ~ "Older \nmales (65+)",
+      variable == "working_age_males" ~ "Working age \nmales (16-64)",
+      variable == "younger_males" ~ "Younger \nmales (< 16)"
+    )
+  ) |>
+  mutate(percent = percent / 100)
```
Comment (MikeJohnPage) [inline data-raw/northern-ireland-hsct-demographics.R:41]: Could we not remove `* 100` on line 13, and drop this line?

File: data-raw/northern-ireland-ltla-demographics.R [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  ) |>
+  pivot_longer(!area_name,
+    names_to = c(".value", "variable"),
+    names_sep = "__"
+  ) |>
+  mutate(
+    variable = case_when(
+      variable == "older_females" ~ "Older \nfemales (65+)",
+      variable == "working_age_females" ~ "Working age \nfemales (20-64)",
+      variable == "younger_females" ~ "Younger \nfemales (< 20)",
```
Comment (MikeJohnPage) [inline data-raw/northern-ireland-ltla-demographics.R:34]: Where did these age ranges come from (e.g., `<20`)?

File: data-raw/northern-ireland-ltla-demographics.R [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      variable == "older_males" ~ "Older \nmales (65+)",
+      variable == "working_age_males" ~ "Working age \nmales (20-64)",
+      variable == "younger_males" ~ "Younger \nmales (< 20)"
+    )
+  ) |>
+  mutate(percent = percent / 100)
+
+# ---- Ethnicity ----
+ethnicity_ni <-
+  ethnicity21_ltla21_ni
```
Comment (MikeJohnPage) [inline data-raw/northern-ireland-ltla-demographics.R:44]: Is there a reason for this renaming? If not, can we remove it and just call `ethnicity21_ltla21_ni` on line 49 below?

File: data-raw/northern-ireland-ltla-demographics.R (hunk 3 above)
Comment (MikeJohnPage) [reply data-raw/northern-ireland-ltla-demographics.R:34]: Why do the ages differ from the HSCT view?

**Review by MikeJohnPage — CHANGES_REQUESTED** [review]

File: data-raw/northern-ireland-ltla-demographics.R (hunk 3 above)
Comment (MatteoLarrode) [reply data-raw/northern-ireland-ltla-demographics.R:34]: The age ranges come from Niuni's data: she found different datasets for local authority and Health level, which use different age thresholds. I checked her data-raw/ files from the demographr package to check the columns she had aggregated to create those categories.

File: data-raw/northern-ireland-hsct-demographics.R (hunk 1 above)
Comment (MatteoLarrode) [reply data-raw/northern-ireland-hsct-demographics.R:30]: Thank you!

**Comment by MikeJohnPage** [comment]
Thanks!

**Pull request merged**